Improve the  validation rules for participationDeadline and votingDea…

Ganja0003 opened the pull request:
What I did:

improved validation for competition deadlines.

participationDeadline must now be on or after startDate and on or before endDate.

votingDeadline must now be on or after participationDeadline and on or before endDate.

siderdk commented:
Great Job Yuusuf, 
I have two comments: 

1- You may want to add minimal client-side checks (e.g. disable the date pickers or show inline errors) so users can’t select a participation deadline before the start date, etc. That just improves UX and prevents them from hitting a backend error only after submitting.  
(I can do this if you want, so you can jump to another task) 

2-the vercel bot shows errors, usually it's just linting and formatting. 
try to rebase, 
npm run lint --fix 
npm run format 

Let me know

siderdk commented:
Oh, one last thing. 
maybe also make the error/feedback messages in Danish?!!

Ganja0003 commented:
I have addressed the comments mentioned, but Im still getting the same error message and failed deployment as the other PR. Here is the error message:  `Error: Missing DOMAIN env var – set DOMAIN=http://localhost:3000 (and in prod to your live URL)
    at 5747 (.next/server/app/api/request-reset/route.js:1:363)
    at t (.next/server/webpack-runtime.js:1:127)
    at t (.next/server/app/api/request-reset/route.js:4:5116)
    at <unknown> (.next/server/app/api/request-reset/route.js:4:5154)
    at t.X (.next/server/webpack-runtime.js:1:1191)
    at <unknown> (.next/server/app/api/request-reset/route.js:4:5129)
    at Object.<anonymous> (.next/server/app/api/request-reset/route.js:4:5181)
> Build error occurred
[Error: Failed to collect page data for /api/request-reset] {
  type: 'Error'
}
Error: Command "npm run build" exited with 1`
